src/Common/src/Interop/Unix/libssl/Interop.libssl.cs

```diff
@@ -109,6 +109,9 @@ internal static partial class libssl
         [DllImport(Interop.Libraries.LibSsl)]
         internal static extern void SSL_CTX_set_quiet_shutdown(SafeSslContextHandle ctx, int mode);
 
+        [DllImport(Interop.Libraries.LibSsl)]
+        internal static extern IntPtr SSL_get_SSL_CTX(SafeSslHandle ssl);
```

Comment:
I think you're returning IntPtr here but wrapping in a SafeHandle elsewhere.  If you need to do that, make this private, and make a helper method.

But you should consider making SafeSharedSslContextHandle : SafeInteriorHandle, and ensuring handle safety.